Source/Core/Core/Movie.cpp

```diff
@@ -432,15 +437,21 @@ void ChangePads()
   {
     if (SerialInterface::SIDevice_IsGCController(SConfig::GetInstance().m_SIDevice[i]))
       controllers |= (1 << i);
+    if (SConfig::GetInstance().m_SIDevice[i] == SerialInterface::SIDEVICE_GC_GBA_EMULATED)
+      controllers |= (1 << (i + 8));
   }
 
-  if ((s_controllers & 0x0F) == controllers)
+  if ((s_controllers & 0x0F0F) == controllers)
```

Comment:
Doing all the masking and bitwise stuff manually seems a bit error prone. Could we add a wrapper to manipulate the controller mask more safely and avoid having to expose those details to non-Movie code?

(I realise the problem already existed prior to your changes, but this PR exacerbates it.)